FriendsWave/social/models.py

```diff
@@ -1,3 +1,90 @@
+from functools import _Descriptor
 from django.db import models
+from django.db.models.base import Model
+from django.db.models.fields.related import OneToOneField
+
+class Profil(models.Models):
+    pseudo = models.CharField(max_length=100)
+    image = models.ImageField()
+    user = models.ForeignKey('auth.User', on_delete=models.CASCADE)
+    friends = models.ManyToManyField('self', through='Friend')
+    topics = models.ManyToManyField('Topic', through='UserTopic')
+    etat = models.BooleanField(default=False)
+
+class Friend(models.Model):
+    follower = models.ForeignKey(Profil, OneToOneField = models.CASCADE)
+    followed = models.ForeignKey(Profil, OneToOneField = models.CASCADE)
+    status = models.BooleanField()
+
+class Topic(models.Model):
+    subject = models.TextField(max_length=100)
+    description = models.TextField(max_length=100)
+    creator = models.ForeignKey(Profil, on_delete=models.SET_NULL)
+
+class UserTopic(models.Model):
+    profil = models.ForeignKey(Profil, on_delete=models.CASCADE)
+    topic = models.ForeignKey(Topic, on_delete=models.CASCADE)
+    choice_date = models.DateField()
+    raison = models.TextField(max_length=100)
+    start_date = models.DateField()
+    end_date = models.DateField(null=True)
+    
+
+class Group(models.Model):
+    designation = models.TextField(max_length=100)
+    _description = models.TextField(max_length=100)
+    type = models.BooleanField()
+    created_at = models.DateField() 
+    members = models.ManyToManyField(Profil, through='Member')
+    private_key = models.TextField(max_length=100, null=True)
+
+class Invitation(models.Model):
+    profil = models.ForeignKey(Profil, on_delete=models.CASCADE)
+    group = models.ForeignKey(Group, on_delete=models.CASCADE)
+    sujbject = models.ForeignKey(Profil, on_delete=models.CASCADE)
+    created_at = models.DateField()
+    etat = models.BooleanField(default=False)
```

Comment:
consider storing etat which might refer to state in a charfield with predefined choice list